CRM/Case/BAO/CaseType.php

```diff
@@ -60,7 +60,13 @@ public static function add(&$params) {
     // function to format definition column
     if (isset($params['definition']) && is_array($params['definition'])) {
       $params['definition'] = self::convertDefinitionToXML($caseTypeName, $params['definition']);
-      CRM_Core_ManagedEntities::scheduleReconciliation();
+      // @see dev/core#3722 Prevent reconciliation from being called infinitely.
+      // @see CRM-14786 Reconcilation added to manage whenever a case-type XML definition changes,
+      // in order to add/remove associated activity and relationship types.
+      if (!isset(\Civi::$statics[__CLASS__][$params['name']])) {
+        \Civi::$statics[__CLASS__][$params['name']] = 1;
```

Comment:
$params['name'] can be blank, e.g. if calling this to update using id instead of name. There is $caseTypeName which should always be set. But I'm wondering if it even needs to be that specific, since (I think) the reconciliation will reconcile whatever needs reconciling and so just needs to be called once. To be honest I'm not sure exactly how it works, and when PHASE_POST_COMMIT occurs.

In reverse, I suppose there's also the situation where, e.g. during upgrade, this might be an issue, because let's say you have something that does casetype.create in 5.52 and something calls reconciliation, then another casetype.create in 5.53 and something calls reconciliation. That second one won't run if running upgrades via cli since the static will stick.

I'm just thinking of some things to test out.

Reply:
Yeah I meant to use `$caseTypeName`.

Based on your second thought it seems like setting it per `$caseTypeName` might be needed. Unless even that could be repeated across upgrades.